crates/libs/windows/src/core/agile_reference.rs

```diff
@@ -0,0 +1,32 @@
+use super::*;
+use bindings::*;
+use core::marker::PhantomData;
+
+pub struct AgileReference<T> {
+    reference: IAgileReference,
+    _marker: PhantomData<T>,
+}
+
+impl<T: Interface> AgileReference<T> {
+    pub fn new<'a>(from_ref: &'a T) -> Result<Self>
+    where
+        &'a T: IntoParam<'a, IUnknown>,
+    {
+        let reference = unsafe { RoGetAgileReference(AGILEREFERENCE_DEFAULT, &T::IID, from_ref)? };
+        Ok(Self { reference, _marker: Default::default() })
```

Comment:
Would it be simpler to use [map](https://doc.rust-lang.org/std/result/enum.Result.html#method.map)?

Reply:
Yeah I didn't see that. Though rustfmt wants to keep it on a single line which makes it a bit harder to read.

Reply:
Unfortunately, I have the rustfmt line width set to 800 to avoid a rustfmt bug.

Reply:
Maybe a good candidate for [`#![rustfmt::skip]`?](https://github.com/rust-lang/rustfmt/blob/master/tests/target/skip.rs)

Reply:
There are just too many places that I would be tempted to apply that. 🤪